Use $(MAKE) instead of raw make for recursion

roystgnr opened the pull request:
Fixes #19619

See
https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable
for the rationale in general, though in this case I'm just fixing a few
spurious warnings.

roystgnr committed:
Use $(MAKE) instead of raw make for recursion

Fixes #19619

See
https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html#MAKE-Variable
for the rationale in general, though in this case I'm just fixing a few
spurious warnings.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/915331/) on 0f55f4d wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/19620/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/915341/) on 0f55f4d wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>a7794a</th>
      <th colspan=4>#19620 0f55f4</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>82.17%</td>
      <td>82.17%</td>
      <td>+0.00%</td>
      <td>-</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>72913</td>
      <td>72914</td>
      <td>+1</td>
      <td>0</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>15821</td>
      <td>15820</td>
      <td>-1</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/19620/coverage_diff/framework.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/19620/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/tensor_mechanics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/19620/coverage/xfem)

</details>



This comment will be updated on new commits.

roystgnr committed:
Remove atavistic comment

John helped me figure this out - in a previous version of the "clobber"
rule, we had a $(shell foo) variable that was actually doing the
removal as a side effect, and those variables get expanded even by "make
-n" - but we're not doing that anymore.

roystgnr commented:
Okay, spurious warnings *and* atavistic comment fixed.  Now this should be ready to merge.

loganharbour commented:
I think the BISON failure is related... and I have no idea why.

roystgnr commented:
Whoa.  Sorry I missed your comment somehow.

Seventeen failures?  All diffs?  From making "cleanall" and "clobberall" aware of the gmake job server???  No way.  I'm just going to kick that test and see what happens.

moosebuild commented:
Job [Internal App tests](https://civet.inl.gov/job/915693/) on b9b837d : invalidated by @roystgnr

Kicking due to dumbfounded disbelief
